File: tests/fixtures/usage-graph-java/com/example/Consumer.java
package com.example;

public class Consumer {
    public int viaInstance() {
        Service s = new Service();
        return s.run();
    }

    public int viaStatic() {
        return Service.helper();
    }

    public Service makeService() {
        return new Service();
    }

    public int shadowed(Service run) {
        // `run` is a Service-typed parameter; `run.run()` resolves to
        // Service.run via the receiver type, not via the method name alone.
        return run.run();
    }

    public int wrongReceiver(Consumer other) {
        // `other` is a Consumer, which has no `run()`; resolving by receiver type
        // gives `Consumer.run` (not a node), so this must NOT edge to Service.run.
        return other.run();
    }

    public int shadowFallback() {
        // `Service` here is an untyped local that merely shares the Service
        // type's name; calling `run()` on it must NOT be reinterpreted as a
        // static call on the `Service` type.
        var Service = 42;
        return Service.run();
    }
}

File: tests/fixtures/usage-graph-csharp/Consumer.cs
namespace Example
{
    public class Consumer
    {
        public int ViaInstance()
        {
            Service s = new Service();
            return s.Run();
        }

        public int ViaStatic()
        {
            return Service.Helper();
        }

        public Service MakeService()
        {
            return new Service();
        }

        public int Shadowed(Service Run)
        {
            // `Run` is a Service-typed parameter; `Run.Run()` resolves to
            // Service.Run via the receiver type, not the member name alone.
            return Run.Run();
        }

        public int WrongReceiver(Consumer other)
        {
            // `other` is a Consumer with no `Run()`; resolves to Consumer.Run
            // (not a node), so this must NOT edge to Service.Run.
            return other.Run();
        }

        public int Local()
        {
            return 7;
        }

        public int CallsLocal()
        {
            // Unqualified call attributes to the enclosing class.
            return Local();
        }
    }
}
